content/blog/2018/egress-https/index.md

```diff
@@ -105,91 +147,194 @@ Now accessing the web page of the application displays the book details without
+{{< text bash >}}
+$ kubectl delete serviceentry googleapis
+$ kubectl delete virtualservice googleapis
+$ kubectl delete -f @samples/bookinfo/networking/virtual-service-details-v2.yaml@
+$ kubectl delete -f @samples/bookinfo/platform/kube/bookinfo-details-v2.yaml@
+{{< /text >}}
 
-{{< image width="80%" ratio="65.16%"
+## TLS origination by Istio
+
+There is a caveat to this story. Suppose you want to monitor which specific set of
+[Google APIs](https://developers.google.com/apis-explorer/) your microservices use
+([Books](https://developers.google.com/books/docs/v1/getting_started),
+[Calendar](https://developers.google.com/calendar/), [Tasks](https://developers.google.com/tasks/) etc.)
+Suppose you want to enforce a policy that using only
+[Books APIs](https://developers.google.com/books/docs/v1/getting_started) is allowed. Suppose you want to monitor the
+book identifiers that your microservices access. For these monitoring and policy tasks you need to know the URL path.
+Consider for example the URL
+[`www.googleapis.com/books/v1/volumes?q=isbn:0486424618`](https://www.googleapis.com/books/v1/volumes?q=isbn:0486424618).
+In that URL, [Books APIs](https://developers.google.com/books/docs/v1/getting_started) is specified by the path segment
+`/books`, and the [ISBN](https://en.wikipedia.org/wiki/International_Standard_Book_Number) number by the path segment
+`/volumes?q=isbn:0486424618`. However, in HTTPS, all the HTTP details (hostname, path, headers etc.) are encrypted and
+such monitoring and policy enforcement by the sidecar proxies are not possible. Istio can only know the server name of
+the encrypted requests by the [SNI](https://tools.ietf.org/html/rfc3546#section-3.1) (_Server Name Indication_) field,
+in this case `www.googleapis.com`.
+
+To allow Istio to perform monitoring and policy enforcement of egress requests based on HTTP details, the microservices
+must issue HTTP requests. Istio then opens an HTTPS connection to the destination (performs TLS origination). The code
+of the microservices must be written differently or configured differently, according to whether the microservice runs
+inside or outside an Istio service mesh. This contradicts the Istio design goal of [maximizing transparency](/docs/concepts/what-is-istio/#design-goals). Sometimes you need to compromise...
+
+The diagram below shows two options of sending HTTPS traffic to external services. On the top, a microservice sends
+regular HTTPS requests, encrypted end-to-end. On the bottom, the same microservice sends unencrypted HTTP requests
+inside a pod, which are intercepted by the sidecar Envoy proxy. The sidecar proxy performs TLS origination, so the
+traffic between the pod and the external service is encrypted.
+
+{{< image width="80%" ratio="95.14%"
```

Comment:
I suggest switching the ratio to 60%. With 80%, the image comes out looking a bit too zoomed in.

Reply:
> I suggest switching the ratio to 60%. With 80%, the image comes out looking a bit too zoomed in.

@geeknoid Do you mean `width` or `ratio`? The current `ratio` is 95%.

Reply:
Sorry, I meant width.

The ratio should be derived from the actual image width & height and is fine as it is.

Reply:
@geeknoid Done, thanks.